docs/decisions/0006_areas_of_responsibility.rst

```diff
@@ -0,0 +1,106 @@
+6. Areas of Responsibility
+###############################
+
+Status
+******
+
+Draft -> Accepted
+
+Context
+*******
+
+The Open Analytics Reference System (OARS) consists of several pieces of technology working together via
+some configuration and scripting. Decisions around where to store data and configuration need to be made
+so that developers understand where to make changes and look for the causes of issues. These decisions
+can have wide ranging impact on things such as extensibility and configurability of the system as a whole.
+
+The guiding principles of this system are:
+
+- Allow each service-based Tutor plugin to run the service separately from OARS, and in different
+  configurations
+- Push opinionated configuration and non-LMS service associations to the OARS plugin
+
+Decisions
+*********
+
+The service-based Tutor plugins (``tutor-contrib-clickhouse``, ``tutor-contrib-ralph``, ``tutor-contrib-superset``)
+will each be able to be run outside the OARS context, such that if members of the community want to use
+them in a modified way, or override OARS defaults, that will be possible. It is still assumed that the
+plugins will be used in an Open edX context and they will integrate themselves with the platform where
+necessary.
+
+Specifically:
+
+#. ``tutor-contrib-clickhouse`` will:
+    #. Create a ClickHouse admin user
+    #. NOT create any databases or tables
+#. ``tutor-contrib-ralph`` will:
+    #. Create an Ralph admin user
+    #. Create a Ralph user for an LMS to use to insert statements
+    #. NOT create any ClickHouse databases, users, or tables
+    #. Note: Ralph will fail to start if the configured ClickHouse database and table
+       for xAPI statements has not yet been created
+#. ``tutor-contrib-superset`` will:
+    #. Create an admin user
+    #. Create the Superset-specific MySQL database and permissions for the admin user (Note:
+       By default this uses the LMS MySQL instance to save on system resources)
+    #. Create a user in LMS for SSO
+    #. Register itself as an SSO target with the LMS
+    #. Disable local user management to force SSO login from LMS
+    #. Create a set of default roles and permissions based on LMS permissions for each user
+       (course staff, global staff, superuser, etc)
+    #. Create row-based filters for queries to enforce course-level permissions (Note:
+       since there are no datasets yet there is nothing to connect them to, but
+       they are created here as they would likely be useful for a non-OARS use case)
+    #. NOT create any ClickHouse databases, users, or tables
+
+The process of tying these plugins together is managed by the ``tutor-contrib-oars`` plugin, which
```

Comment:
I'm not very familiar with the tutor plugins and am curious, is it possible to enable the tutor plugin from the other tutor plugin? If we enable `tutor-contrib-oars` could it be configured to install/enable `tutor-contrib-clickhouse`, `tutor-contrib-ralph`, and `tutor-contrib-superset`?

Reply:
@idegtiarov I don't think that functionality exists right now. I think it would complicate the case when someone wants to host one of the services outside of Tutor, for instance using ClickHouse Cloud for hosting the database. I don't have a great idea of how much that use case, or the case of swapping out various services in general, matter to the community, though. 

Right now, if someone wanted to use ClickHouse Cloud they could just not install the `tutor-contrib-clickhouse` plugin and update the Tutor configuration manually to point to their instance and the OARS scripts should work. In theory, at least, it hasn't been tried yet. :)